src/controllers/articleController.js

```diff
@@ -81,15 +81,75 @@ class ArticleController {
         ]
       });
       if (!articles.length) {
-        return res.status(200).json({
+        return res.status(404).json({
           message: 'No articles found at the moment! please come back later'
         });
       }
       res.status(200).json({
         articles
       });
     } catch (err) {
-      throw (err);
+      throw err;
+    }
+  }
+
+  /**
+   * @description update article
+   * @param {object} req
+   * @param {object} res
+   * @return {object} return object with updated article.
+   */
+  static async updateArticle(req, res) {
+    try {
+      const article = {};
+      const {
+        title, description, body, tags, category
+      } = req.body;
+      const originalArticle = await Article.findOne({ where: { slug: req.userData.slug } });
+      if (title) {
+        article.title = title.trim();
+        article.slug = slugGen(title.trim());
+      }
+      article.description = (description || originalArticle.description).trim();
```

Comment:
`
article.description = description.trim() || originalArticle.description;
`

Reply:
I also see that trim has been added to the validation rules, is there a reason we have it here as well?

Reply:
If the description is undefined it throws an error "can not trim value of undefined"

Reply:
> I also see that trim has been added to the validation rules, is there a reason we have it here as well?

In the validations am checking for empty fields but in the controller am trimming white spaces before or after a string.

Reply:
`article.description = description ? description.trim() : originalArticle.description;`